File: Java/RefactoringWorkshop/src/test/java/ooabusers/switches/CarTest.java
package ooabusers.switches;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.*;

public class CarTest {

    Car sportsCar;
    Car familyCar;
    Car smallCar;
    Car defaultCar;
    @Before public void
    setup() {
        sportsCar = new Car(Car.SPORTS);
        familyCar = new Car(Car.FAMILY);
        smallCar = new Car(Car.SMALL);
        defaultCar = new Car(Car.DEFAULT);
    }

    @Test public void
    should_return_the_fuel_capacity_of_the_sports_car() {
        assertEquals("100 liters", sportsCar.fuelCapacity());
    }

    @Test public void
    should_return_the_fuel_capacity_of_the_family_car() {
        assertEquals("40 liters", familyCar.fuelCapacity());
    }

    @Test public void
    should_return_the_fuel_capacity_of_the_small_car() {
        assertEquals("20 liters", smallCar.fuelCapacity());
    }

    @Test public void
    should_return_the_fuel_capacity_of_the_default_car() {
        assertEquals("50 liters", defaultCar.fuelCapacity());
    }

    @Test public void
    should_return_the_fuel_average_of_the_sports_car() {
        assertEquals("3 Km/Liter", sportsCar.fuelAverage());
    }

    @Test public void
    should_return_the_fuel_average_of_the_family_car() {
        assertEquals("10 Km/Liter", familyCar.fuelAverage());
    }

    @Test public void
    should_return_the_fuel_average_of_the_small_car() {
        assertEquals("15 Km/Liter", smallCar.fuelAverage());
    }

    @Test public void
    should_return_the_fuel_average_of_the_default_car() {
        assertEquals("12 Km/Liter", defaultCar.fuelAverage());
    }

    @Test public void
    should_return_the_features_of_the_sports_car() {
        assertEquals("Sports car limited edition", sportsCar.features());
    }

    @Test public void
    should_return_the_features_of_the_family_car() {
        assertEquals("Best family car in the segment", familyCar.features());
    }

    @Test public void
    should_return_the_features_of_the_small_car() {
        assertEquals("Very small car", smallCar.features());
    }

    @Test public void
    should_return_the_features_of_the_default_car() {
        assertEquals("default car", defaultCar.features());
    }
}

File: C#/RefactoringWorkshop/Test/OOAbusers/Switches/CarTest.cs
using NUnit.Framework;
using RefactoringWorkshop.OOAbusers.Switches;

namespace Test.OOAbusers.Switches
{
    public class CarTest
    {
        private ICar _sportsCar;
        private ICar _familyCar;
        private ICar _smallCar;
        private ICar _defaultCar;

        [SetUp]
        public void Setup() {
            _sportsCar = new Car(Car.SPORTS);
            _familyCar = new Car(Car.FAMILY);
            _smallCar = new Car(Car.SMALL);
            _defaultCar = new Car(Car.DEFAULT);
        }

        [Test]
        public void should_return_the_fuel_capacity_of_the_sports_car() {
            Assert.AreEqual("100 liters", _sportsCar.FuelCapacity());
        }

        [Test]
        public void should_return_the_fuel_capacity_of_the_family_car() {
            Assert.AreEqual("40 liters", _familyCar.FuelCapacity());
        }

        [Test]
        public void should_return_the_fuel_capacity_of_the_small_car() {
            Assert.AreEqual("20 liters", _smallCar.FuelCapacity());
        }

        [Test]
        public void should_return_the_fuel_capacity_of_the_default_car() {
            Assert.AreEqual("50 liters", _defaultCar.FuelCapacity());
        }

        [Test]
        public void should_return_the_fuel_average_of_the_sports_car() {
            Assert.AreEqual("3 Km/Liter", _sportsCar.FuelAverage());
        }

        [Test]
        public void should_return_the_fuel_average_of_the_family_car() {
            Assert.AreEqual("10 Km/Liter", _familyCar.FuelAverage());
        }

        [Test]
        public void should_return_the_fuel_average_of_the_small_car() {
            Assert.AreEqual("15 Km/Liter", _smallCar.FuelAverage());
        }

        [Test]
        public void should_return_the_fuel_average_of_the_default_car() {
            Assert.AreEqual("12 Km/Liter", _defaultCar.FuelAverage());
        }

        [Test]
        public void should_return_the_features_of_the_sports_car() {
            Assert.AreEqual("Sports car limited edition", _sportsCar.Features());
        }

        [Test]
        public void should_return_the_features_of_the_family_car() {
            Assert.AreEqual("Best family car in the segment", _familyCar.Features());
        }

        [Test]
        public void should_return_the_features_of_the_small_car() {
            Assert.AreEqual("Very small car", _smallCar.Features());
        }

        [Test]
        public void should_return_the_features_of_the_default_car() {
            Assert.AreEqual("default car", _defaultCar.Features());
        }
    }
}